packages/plugins/plugin-debug/src/DebugPlugin.tsx

```diff
@@ -30,25 +21,32 @@ import {
   topbarBlockPaddingStart,
 } from '@dxos/react-ui-theme';
 
-import { DebugGlobal, DebugSettings, DebugSpace, DebugObjectPanel, DebugStatus, Wireframe } from './components';
+import { DebugGlobal, DebugObjectPanel, DebugSettings, DebugSpace, DebugStatus, Wireframe } from './components';
 import meta, { DEBUG_PLUGIN } from './meta';
 import translations from './translations';
-import { DebugContext, type DebugSettingsProps, type DebugPluginProvides, DebugAction } from './types';
-
-export const SETTINGS_KEY = DEBUG_PLUGIN + '/settings';
+import {
+  DebugAction,
+  DebugContext,
+  type DebugPluginProvides,
+  type DebugSettingsProps,
+  DebugSettingsSchema,
+} from './types';
 
-export const DebugPlugin = (): PluginDefinition<DebugPluginProvides> => {
-  const settings = new LocalStorageStore<DebugSettingsProps>(DEBUG_PLUGIN, { debug: true, devtools: true });
-  let intentPlugin: Plugin<IntentPluginProvides>;
+export const DebugPlugin = definePlugin<DebugPluginProvides>((context) => {
+  const settings = create<DebugSettingsProps>({
+    debug: true,
+    devtools: true,
+  });
 
   return {
     meta,
     ready: async (plugins) => {
-      intentPlugin = resolvePlugin(plugins, parseIntentPlugin)!;
-      settings
-        .prop({ key: 'debug', type: LocalStorageStore.bool({ allowUndefined: true }) })
-        .prop({ key: 'devtools', type: LocalStorageStore.bool({ allowUndefined: true }) })
-        .prop({ key: 'wireframe', type: LocalStorageStore.bool({ allowUndefined: true }) });
+      context.init(plugins);
+      context.resolvePlugin(parseSettingsPlugin).provides.settingsStore.createStore({
```

Comment:
I realize you're trying to remove the plugin variables, but I think I would find this easier to read if `context.resolvePlugin(parseSettingsPlugin)` was assigned to a variable, this is a lot